index.js

```diff
@@ -216,16 +222,19 @@ function InigoPlugin(config) {
   }
 
   // instance stored in a closure
-  let instance = 0;
   if (config.Schema) {
-    instance = new Inigo(config);
+    if (rootInigoInstance !== 0) {
+      throw new Error("Only one instance of InigoPlugin can be created.")
+    }
+
+    rootInigoInstance = new Inigo(config);
```

Comment:
I don't consider a good approach to restrict the number of instances to 1 considering that we have a semaphore available in the bindings to acquire the shared pipeline which resolves this kind of issues.

Reply:
Definitely will rework this part. Considering the timeline, I was thinking of changing this after the prod push.